### PR TITLE
Hack day/multi step progress bar

### PR DESCRIPTION
**Pull request by zghaed**
Added MultiStepProgressItem and MultiStepProgressBar components
and their tests and demos to canon-react.

screen shot 2016-08-12 at 1 42 39 pm

### CODE REVIEW

File: src/MultiStepProgressBar.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  render() {
+    const sizeClass = classNames(
+      'rs-progress',
+      SIZE_CLASSES[this.props.size]
+    );
+
+    return (
+      <div className={ sizeClass }>
+        <div className="rs-progress-inner">
+          {this.props.children}
```
Comment (hannahstwinter) [inline src/MultiStepProgressBar.jsx]: spaces - ```{ this.props.children }```

File: src/MultiStepProgressItem.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  render() {
+    const statusClass = classNames(
+      'rs-bar',
+      STATUS_CLASSES[this.props.status],
+      TYPE_CLASSES[this.props.type]
+    );
+    const style = { 'width': `${ this.props.progress }%` };
+
+    return (
+      <div className="rs-segment" style={ style }>
```
Comment (hannahstwinter) [inline src/MultiStepProgressItem.jsx]: Had you tried using the existing ProgressBar component here? I'm curious as to whether that would work... There's a fair amount of duplication here...

Comment (hannahstwinter) [inline src/MultiStepProgressItem.jsx]: If it's not reasonable we should at least pull out the shared class maps.

Comment (zghaed) [inline src/MultiStepProgressItem.jsx]: Yeah, I know there's a fair amount of duplication here. MultiStepProgressItem has `caption` which ProgressBar doesn't.

Comment (zghaed) [inline src/MultiStepProgressItem.jsx]: @hannahstwinter I can add an optional caption to ProgressBar and remove this component. What do you think?

**Comment by jdrush89** [comment]
Check out the commit guidelines https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#-git-commit-guidelines
You should do a rebase -i and squash to one commit, something like:
feat: adding MultiStepProgressItem and MultiStepProgressBar

**Pull request closed without merging**